### DISCUSSION TITLE
Improve UML

### DISCUSSION THREAD

**Discussion by mattiacolombomc** [PiervitoCreanza/IS24-AM02#7] [Ideas]
- [x] Questione dei corner: chiamo getGameGameObject() e getGameResource() su tutti i corner per capire il tipo EFFETTIVO di corner in base a risultato chiamata?
- [x] good practice avere getter "aggregato" di elementi delle sottoclassi (es. un possibile che metodo che ritorna ogni corner di una PlaceableCard con il tipo giusto, quindi o GameObject o GameResource?
- [x] sono over-engineered le sottoclassi dei Corner?
- [x] Front e Back della carta?
- [x] Questione del side: per salire di un livello da PlaceableSide per avere Side, come faccio a far convivere questa cosa con StarterCard?

**Comment by PiervitoCreanza** [comment]
- [ ]  Dove salviamo il deck iniziale?

**Comment by mattiacolombomc** [comment]
- [x] possibile che non esista un metodo per sottolineare i metodi utilizzabili da composizione e quelli invece ereditati da sottoclassi?

velocizzerebbe molto il tutto, evitando cose del genere...
_(Seduta stante l'errore fatto con UML non ognuno sui propri branch, dal quale abbiamo imparato)_
![CleanShot 2024-03-18 at 22 56 02](https://github.com/PiervitoCreanza/IS24-AM02/assets/55593934/56eda1f6-508b-4b60-8a0a-20c2330aa0f9)
